File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/AcidSlime_M_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";


export class AcidSlime_M_Model extends MonsterInfo{
    public static Id = "AcidSlime_M"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(29, 34);
        else
            return Mathf.RandomRange(28, 32);
    }
    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/slimeM/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/slimeM/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/slimeM/skeleton.png"));
    }

    get Name(): string {
        return "酸液史莱姆（中）";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/AcidSlime_M_Model.cs
using ModTheSpire.Dungeon;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class AcidSlime_M_Model : MonsterInfo
    {
        public static string Id = "AcidSlime_M";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(29, 34) : Random.Range(28, 32);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/slimeM/skeleton.atlas",
                    "Monsters/theBottom/slimeM/skeleton.json", "Monsters/theBottom/slimeM/skeleton.png");
            }
        }

        public override string Name => "酸液史莱姆（中）";

        public AcidSlime_M_Model(Vector2 pos) : base(pos)
        {
        }
    }
}